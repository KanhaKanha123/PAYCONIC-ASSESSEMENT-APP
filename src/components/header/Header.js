import { Container, Navbar, Nav } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import { RoutesConfig } from '../../constants/constant';
import { CurrencyExchangeAppState } from '../../store/context/Context';

const Header = () => {
    const navigate = useNavigate();

    //Global application state for exchange details results
    const { dispatchExchangeHistory } = CurrencyExchangeAppState();

    //navigate to dashboard page
    const navigateClickExchange = () => {
        //clear history state when manually click on navigation
        dispatchExchangeHistory({ type: "CURRENCY_EXCHANGE_INPUT", payload: {} });
        navigate(RoutesConfig.dashboard);
    };

    //navigate to history page
    const navigateClickHistory = () => {
        navigate(RoutesConfig.history);
    };

    return (<Navbar aria-label='exchange app header' data-testid="header" bg='dark' variant='dark' className='app-header'>
        <Container>
            <Navbar.Brand className='flex-header--item-left'>
                <div className="tab">
                    <button onClick={navigateClickExchange} className="tablinks">Exchange</button>
                    <button onClick={navigateClickHistory} className="tablinks">History</button>
                </div>
            </Navbar.Brand>
            <Navbar.Brand className='flex-header--item-right'>
            </Navbar.Brand>
        </Container>
        <Nav></Nav>
    </Navbar>);
}

export default Header;