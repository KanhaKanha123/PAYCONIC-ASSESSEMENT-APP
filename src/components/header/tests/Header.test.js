import { render, screen, fireEvent, act, waitFor } from '@testing-library/react';
import Header from '../Header';
import renderer from 'react-test-renderer';

describe('Header component test', () => {
    const mockSetState = jest.fn();

    test("Header component load correctly", () => {

        render(<Header />);

        const header = screen.getAllByTestId('header');
        const headerText = screen.getAllByText(/TODO APP/i);
        const headerButton = screen.getAllByTestId('header-btn');

        // check element exist
        expect(header).toBeInTheDocument;
        expect(headerText).toBeInTheDocument;
        expect(headerButton).toBeInTheDocument;
    });

    test("addToDo method should be called", async () => {

        await act(() => { render(<Header />) });

        const headerButton = screen.getAllByTestId('header-btn');

        waitFor(() => {
            fireEvent.click(headerButton, { id: 1, value: 'check' })
            expect(mockSetState).toHaveBeenCalled;
        });
    });

    test("Header component should match snapshot", async () => {
        const element = await renderer.create(<Header />);
        expect(element.toJSON()).toMatchSnapshot();
    });
});
