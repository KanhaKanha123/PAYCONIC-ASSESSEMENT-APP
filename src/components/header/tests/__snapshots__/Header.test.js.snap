// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test Header component should match snapshot 1`] = `
<nav
  aria-label="tv show header"
  className="app-header navbar navbar-expand navbar-dark bg-dark"
  data-testid="header"
>
  <div
    className="container"
  >
    <span
      className="flex-header--item-left navbar-brand"
    >
      <span>
        TODO APP
      </span>
    </span>
    <span
      className="flex-header--item-right navbar-brand"
    >
      <button
        className="button"
        data-testid="header-btn"
        onClick={[Function]}
      >
        Add TODO
      </button>
    </span>
  </div>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  />
</nav>
`;
